models/actions/run_job_list.go

```diff
@@ -21,7 +22,33 @@ func (jobs ActionJobList) GetRunIDs() []int64 {
 	})
 }
 
+func (jobs ActionJobList) LoadRepos(ctx context.Context) error {
+	repoIDs := container.FilterSlice(jobs, func(j *ActionRunJob) (int64, bool) {
+		return j.RepoID, j.RepoID != 0
```

Comment:
Per if in line 38, maybe not worth it to change
```suggestion
		return j.RepoID, j.RepoID != 0 && j.Repo == nil
```